src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -200,6 +200,47 @@ void ScriptedAI::DoPlaySoundToSet(WorldObject* source, uint32 soundId)
     source->PlayDirectSound(soundId);
 }
 
+void ScriptedAI::AddThreat(Unit* victim, float amount, Unit* who)
+{
+    if (!who)
+        who = me;
+    who->AddThreat(victim, amount, nullptr, true, true);
```

Comment:
Would like some consistency here, some of these functions are on Unit (AddThreat, ResetThreatList) and some are on ThreatManager (ModifyThreatByPercent, ResetThreat, GetThreat)